kgforge/specializations/databases/utils.py

```diff
@@ -0,0 +1,88 @@
+#
+# Blue Brain Nexus Forge is free software: you can redistribute it and/or modify
+# it under the terms of the GNU Lesser General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# Blue Brain Nexus Forge is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU Lesser
+# General Public License for more details.
+#
+# You should have received a copy of the GNU Lesser General Public License
+# along with Blue Brain Nexus Forge. If not, see <https://choosealicense.com/licenses/lgpl-3.0/>.
+
+import os
+import json
+import requests
+from typing import Optional, Any
+
+from kgforge.core import Resource
+from kgforge.core.commons.exceptions import DownloadingError, QueryingError
+
+
+def type_from_filters(*filters) -> Optional[str]:
+    """Returns the first `type` found in filters."""
+    resource_type = None
+    filters = filters[0]
+    if isinstance(filters, dict):
+        if 'type' in filters:
+            resource_type = filters['type']
+    else:
+        # check filters grouping
+        if isinstance(filters, (list, tuple)):
+            filters = [filter for filter in filters]
+        else:
+            filters = [filters]
+        for filter in filters:
+            if 'type' in filter.path and filter.operator is "__eq__":
+                resource_type = filter.value
+                break
+    return resource_type
+
+def resources_from_results(results):
```

Comment:
Similarly to the construct one https://github.com/BlueBrain/nexus-forge/blob/865ed625b843acd110c9a84be9fbc4fc630a3cba/kgforge/core/commons/sparql_query_builder.py#L234